File: src/main/java/com/saomc/saoui/effects/RenderDispatcher.java
package com.saomc.saoui.effects;

import com.saomc.saoui.GLCore;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.Tessellator;
import net.minecraft.profiler.Profiler;
import org.lwjgl.opengl.GL11;

/**
 * This code was original created by <Vazkii> and has been modified to our needs
 * All credit goes to him
 */
@SideOnly(Side.CLIENT)
public final class RenderDispatcher {

    public static int particleFxCount = 0;

    public static void dispatch() {
        Tessellator tessellator = Tessellator.instance;

        Profiler profiler = Minecraft.getMinecraft().mcProfiler;

        //GL11.glPushAttrib(GL11.GL_LIGHTING);
        GLCore.glBlend(true);
        GLCore.blendFunc(GL11.GL_ONE, GL11.GL_ONE);

        profiler.startSection("death particle");
        DeathParticles.dispatchQueuedRenders(tessellator);
        profiler.endSection();
        GLCore.blendFunc(GL11.GL_ONE, GL11.GL_ZERO);
        GLCore.glBlend(false);
        //GL11.glPopAttrib();
    }
}

File: src/main/java/com/saomc/saoui/effects/RenderDispatcher.kt
/*
 * Copyright (C) 2016-2019 Arnaud 'Bluexin' Solé
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.saomc.saoui.effects

import com.saomc.saoui.GLCore
import net.minecraft.client.Minecraft
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import org.lwjgl.opengl.GL11

/**
 * This code was original created by <Vazkii> and has been modified to our needs
 * All credit goes to him
</Vazkii> */
@SideOnly(Side.CLIENT)
object RenderDispatcher {

    var particleFxCount = 0

    fun dispatch() {
        val profiler = Minecraft.getMinecraft().profiler

        GLCore.glBlend(true)
        GLCore.blendFunc(GL11.GL_ONE, GL11.GL_ONE)
        GLCore.lighting(false)
        profiler.startSection("death particle")
        DeathParticles.dispatchQueuedRenders()
        profiler.endSection()
        GLCore.blendFunc(GL11.GL_ONE, GL11.GL_ZERO)
        GLCore.glBlend(false)
    }
}
